pkg/server/server.go

```diff
@@ -238,6 +240,19 @@ func (s *Server) Run(ctx context.Context) error {
 		return apiHandler
 	}
 
+	// Update the generic config external address based on the secure serving default port,
+	// so its complete representation is provided to the external address initializer.
+	if _, _, err := net.SplitHostPort(genericConfig.ExternalAddress); err != nil {
+		if genericConfig.SecureServing == nil {
+			klog.Fatalf("cannot derive external address port without listening on a secure port.")
+		}
+		_, port, err := genericConfig.SecureServing.HostPort()
+		if err != nil {
+			klog.Fatalf("cannot derive external address from the secure port: %v", err)
+		}
+		genericConfig.ExternalAddress = net.JoinHostPort(genericConfig.ExternalAddress, strconv.Itoa(port))
```

Comment:
is this a bug in kube?

Reply:
I don't think so. It's the admission plugin that's initialised in KCP before the `Complete` method is actually called on the generic options.